### ISSUE TITLE
Keras: How to use fit_generator with 2 input images which corresponds to single output Image

### ISSUE THREAD

**Issue by ramatubs** [keras-team/keras#14701]
I'm creating a CNN and I want to use as input a picture height 40, width 120, and 2 channels, and on the output height 40, width 120, and 1 channel

      `def generate_data_generator_for_two_images(genX1, genX2, genY):
          while True:
              X1i = genX1.next() #input img1
              X2i = genX2.next() #input img2
              Y = genY.next() #Lable (output)
              yield [X1i[0], X2i[0]], Y[0]
      
          train_inimggen = train_inimggen.flow_from_directory(train_closeconv_dir, target_size=(40, 120), color_mode='grayscale',
                                                          class_mode=None,
                                                          seed=250, batch_size=2, shuffle=False)
          train_gradimggen = train_gradimggen.flow_from_directory(train_grad_dir, target_size=(40, 120), color_mode='grayscale',
                                                              class_mode=None,
                                                              seed=250, batch_size=2, shuffle=False)
          train_finalimggen = train_finalimggen.flow_from_directory(train_final_dir, target_size=(40, 120),
                                                                color_mode='grayscale', class_mode=None,
                                                                seed=250, batch_size=2, shuffle=False)
          model = CNN.build()
      
      
          gen_flow = generate_data_generator_for_two_images(train_inimggen, train_gradimggen, train_finalimggen)
      
          model.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['accuracy'])
      
          model.fit_generator(gen_flow, steps_per_epoch=2, epochs=5)`

Following is my Input layer to the CNN

          `def build():
                input_img = Input(shape=(40,120,2), name='input_tensor1')
               conv11 = Conv2D(64, (3, 3), strides=(1, 1), padding='same', activation='relu')(input_img)
                .....
               model = Model(input_img, output_img)
               return model`

I am getting the following error 

Error Message - 

tensorflow.python.framework.errors_impl.invalidargumenterror: incompatible shapes: [40,40,120] vs [40,120,1]

Please help me resolving the issue

**Comment by shubham0204** [comment]
You can probably try implementing your own data loading pipeline using `tf.data.Dataset`. See [this](https://www.tensorflow.org/tutorials/load_data/images#using_tfdata_for_finer_control) documentation to get started. Also, for image augmentation I'd suggest you to explore the [`tf.image.*`](https://www.tensorflow.org/api_docs/python/tf/image) and [`tf.io.*`](https://www.tensorflow.org/api_docs/python/tf/io) modules which contain various helpful methods to read, convert and transform images.

**Comment by rmothukuru** [comment]
@ramatubs,
[**`model.fit_generator`** ](https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit_generator) is deprecated in favor of [**`model.fit`**](https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit).

You can use [**`model.fit`**](https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit) with **`x`** as **`[X1i[0], X2i[0]]`** and **`y`** as **`Y[0]`**.

**Comment by rmothukuru** [comment]
@ramatubs,
This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you.

**Comment by rmothukuru** [comment]
Closing as stale. Please reopen if you'd like to work on this further.
